### PR TITLE
Cfin 237 meta collection config

### PR DESCRIPTION
**Pull request by RichardAllum**
This request contains the changes to the custom gather groovy script as well as the config files needed for the super/meta collection to work.

### CODE REVIEW

File: config/courses/course-search.ftl [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        "results":
+        [
+        <@s.Results>
+            <#if s.result.class.simpleName != "TierBar"><#-- TierBar is a sort of sub-heading, so we need to ignore any -->
+            <#-- EACH RESULT -->
+                {
+                "title":
+                <#if s.result.metaData.courseTitle??>
+                    "${s.result.metaData.courseTitle?json_string}"
+                <#elseif s.result.metaData.t??>
```
Comment (RichardAllum) [inline config/courses/course-search.ftl:22]: There's actually a change here, in `york-uni-courses` this is `s.result.metadata.t?json_string??` but the HYMS collection doesn't have a `t` metadata field, so this is more useful.

**Comment by FergusMcGlynn** [comment]
I'm a bit confused - a number of the files in the PR seem to also be [in another PR](https://github.com/university-of-york/uoy-config-funnelback-courses/pull/37) - is this intentional? Does one of these PRs take precedence? Apologies if I've missed something that's already been discussed.

File: config/external-courses/custom_gather.groovy [hunk 2]
```diff
@@ -1,46 +1,42 @@
...
-    // println "course details for record " + i + " are " + course
+def numRecords = coursesJson.courses.size()
+println "Number of courses is " + numRecords
 
+coursesJson.courses.eachWithIndex { course, i ->
     //skip empty records, there is always a false record at the end of the CMS output
-    if(!course) {
-        i++
+    if (!course) {
+        println "Record " + i + " of " + numRecords + " is false, skipping..."
```
Comment (ChristianBeddows) [inline config/external-courses/custom_gather.groovy:39]: This could be a bit nicer as a full GString e.g. `"Record $i of ${numRecords}..."

Comment (ChristianBeddows) [reply config/external-courses/custom_gather.groovy:39]: Same for the rest of the occurrences in this file

**Review by ChristianBeddows — APPROVED** [review]

**Review by ce666-york — APPROVED** [review]

**Comment by RichardAllum** [comment]
> I'm a bit confused - a number of the files in the PR seem to also be [in another PR](https://github.com/university-of-york/uoy-config-funnelback-courses/pull/37) - is this intentional? Does one of these PRs take precedence? Apologies if I've missed something that's already been discussed.

This PR is primarily for the new meta collection config, the other one was set up as a full branch PR and so contains everything. These other requests are to merge into the main 237 branch, so once these are approved the content in CFIN-237 will be able to merge into main. Sorry for any confusion!

**Review by FergusMcGlynn — APPROVED** [review]

**Pull request closed without merging**